app/scripts/lib/metaRPCClientFactory.js

```diff
@@ -10,6 +11,21 @@ class MetaRPCClient {
     this.requests = new Map();
     this.connectionStream.on('data', this.handleResponse.bind(this));
     this.connectionStream.on('end', this.close.bind(this));
+    this.responseHandled = {};
+  }
+
+  send(id, payload, cb) {
+    this.requests.set(id, cb);
+    this.connectionStream.write(payload);
+    this.responseHandled[id] = false;
+    setTimeout(() => {
+      if (!this.responseHandled[id] && cb) {
+        return cb(new Error('No response from RPC'), null);
+      }
+
```

Comment:
we should clean up `this.responseHandled[id]` and in the error case.

```suggestion
      delete this.responseHandled[id];
```

Reply:
Fixed [here](https://github.com/MetaMask/metamask-extension/pull/14461/commits/a0f0705168558eae4532761b74529f6c01b0e086)